### DISCUSSION TITLE
Linux DRM and vnc server

### DISCUSSION THREAD

**Discussion by dedounet** [AvaloniaUI/Avalonia#12980] [Q&A]
Avalonia 0.10.19
.Net 7
Rapbian Lite x64
Hi everyone, I'm working on an app using an dedicated screen and touchscreen. I've been testing the support of https://github.com/ponty/framebuffer-vncserver which is doing ok with the `builder.StartLinuxFbDev(args);` using the fb0.
Like this I'm accessing the framebuffer cast with RealVNC with no problem. 

My problem is when I switch to the DRM renderer : `builder.StartLinuxDrm(args);` . I assume that the drm system is using /dev/dri/card0 to render the layer on the screen. BUT I can't find a way to see the application on VNC : when I use the fb0 to connect I only see the tty1, not the application. It's not possible to connect anything else than a the fb0 with framebuffer-vncserver.

I also have tested tigervnc-server and tightvncserver but they are dedicated to use with an X server and are not usefull with drm.
Does anyone have an hint ?

**Comment by Coloryr** [comment]
```
BuildAvaloniaApp()
    //.StartLinuxFbDev(args);
    .StartLinuxDrm(args, card: "/dev/dri/card0", scaling: 1);
```

**Reply by dedounet** [reply]
Well  thanks for the proposal but my problem is around the vnc server : I can't find a device to stream and access the launched application by drm. I have no problem to launch the app with my code.
I'll try to be clear :
When I start the app with framebuffer, I can access the fb0 from framebuffer-vncserver. 
When I start the app with drm, I can't start the framebuffer-vncserver on /dev/dri/card0 and if i access to fb0 I only see the background console, not the app.
I don't want to use tightvncserver since it's starting an X display.

**Comment by workgroupengineering** [comment]
Have you tried https://github.com/isjerryxiao/kmsvnc? It is in an early stage.

**Reply by dedounet** [reply]
I've been checking this also, impossible to compile ... :) I'll try again, thanks

**Comment by pedroveiga99** [comment]
Hey, @dedounet I'm experiencing the same. Did you ever found out a solution?

**Comment by dedounet** [comment]
Hi there, no, and I'm not working anymore on this project... But a workaround was to use the remote use of avalonia. Then I gave up using a non desktop distrib, sorry pal, I understand your despair and I sympathize!

**Reply by pedroveiga99** [reply]
What do you mean with "remote use of avalonia" ?

**Reply by dedounet** [reply]
Well I was launching my app through SSH remotely and if the screen was not specified, it was displayed on my PC (I was using Windterm) using the resources of the remote machine. By using the Avalonia debugging tools and attaching the debugger a posteriori, I got somewhere.

**Reply by pedroveiga99** [reply]
Oh I think I get it. But yeah that does not work for me. Thanks anyway

**Reply by dedounet** [reply]
As I remember, I think I gaveway and used framebufferVncserver with the fbVersion for everyday work and switched to DRM version to check what was broken ... And It was a pain. 
Good luck with that !
